source/TextDatasource.js

```diff
@@ -163,6 +163,20 @@ class TextDatasource {
     toString() {
         return JSON.stringify(this.toObject());
     }
+
+    /**
+     * Creates a hash ID based on path
+     * @returns {String} an id based on path 
+     * @memberof TextDataSource
+     */
+    hashId(path) {
+        let id = 0
+        for (i = 0; i < path.length; i++) {
+            let ascii_code = path.charCodeAt(i)
+            id += 128 * id + ascii_code
```

Comment:
What is `path`? A text datasource can come from **anywhere**, so unfortunately the only data we have to create a repeatable ID from is the _content_. A hash of the content could work very well.. Why not use [`hash.js`](https://www.npmjs.com/package/hash.js) to hash the content (SHA256 should be fine) and return the hash value as the ID?

`this._content` has the content. An error should be thrown if the content is empty.